src/components/decimal/decimal.test.tsx

```diff
@@ -0,0 +1,1406 @@
+    render(<ControlledDecimal startingValue="1" onBlur={onBlur} />);
+
+    const decimalInput = screen.getByRole("textbox");
+    decimalInput.focus();
+    await user.keyboard("{Backspace}");
+    await user.tab();
+
+    expect(decimalInput).toHaveValue("0.00");
+    expect(screen.getByTestId("hidden-input")).toHaveValue("0.00");
+    expect(onBlur).toHaveBeenCalledWith(
+      expect.objectContaining({
+        target: {
+          value: {
+            formattedValue: "0.00",
+            rawValue: "0.00",
+          },
+        },
+      })
+    );
+  });
+});
+
+describe("refs", () => {
+  it("accepts ref as a ref object", () => {
+    const ref = { current: null };
+    render(<Decimal ref={ref} />);
+
+    expect(ref.current).toBe(screen.getByRole("textbox"));
+  });
+
+  it("accepts ref as a ref callback", () => {
+    const ref = jest.fn();
+    render(<Decimal ref={ref} />);
+
+    expect(ref).toHaveBeenCalledWith(screen.getByRole("textbox"));
+  });
+
+  it("sets ref to empty after unmount", () => {
+    const ref = { current: null };
+    const { unmount } = render(<Decimal ref={ref} />);
+
+    unmount();
+
+    expect(ref.current).toBe(null);
+  });
+});
+
+test("when wrapped in an I18nProvider, the appropriate locale is used, and the formatting updates on blur when the locale is changed", async () => {
+  const user = userEvent.setup();
+  const { rerender } = render(
+    <I18nProvider
+      locale={{
+        locale: () => "fr-FR",
+      }}
+    >
+      <Decimal />
+    </I18nProvider>
+  );
+  expect(screen.getByRole("textbox")).toHaveValue("0,00");
+
+  rerender(
+    <I18nProvider
+      locale={{
+        locale: () => "en-GB",
+      }}
+    >
+      <Decimal />
+    </I18nProvider>
+  );
+  screen.getByRole("textbox").focus();
+  await user.tab();
+  expect(screen.getByRole("textbox")).toHaveValue("0.00");
+});
+
+test("the `id` prop is passed to the input element", () => {
+  render(<Decimal id="decimalId" />);
+
+  expect(screen.getByRole("textbox")).toHaveAttribute("id", "decimalId");
+});
+
+test("no error message is triggered if the `precision` prop is not specified", () => {
+  const consoleSpy = jest
+    .spyOn(global.console, "error")
+    .mockImplementation(() => {});
+
+  render(<Decimal defaultValue="12345.654" />);
+  expect(consoleSpy).not.toHaveBeenCalled();
+
+  consoleSpy.mockRestore();
+});
+
+test("the `required` prop is passed to the visible input and adds an asterisk to the label", () => {
+  render(<Decimal label="required" required />);
+
+  expect(screen.getByRole("textbox")).toBeRequired();
+  expect(screen.getByText("required")).toHaveStyleRule("content", '"*"', {
+    modifier: "::after",
+  });
+});
+
+test("the `required` prop is not passed to the hidden input", () => {
+  render(<Decimal label="required" required />);
+
+  expect(screen.getByTestId("hidden-input")).not.toBeRequired();
+});
+
+test("the `isOptional` prop adds an '(optional)' suffix to the label", () => {
```

Comment:
**suggestion:** Unless this is for coverage and a second reviewer agrees, I think you could move this test to be a Chromatic example as it's testing styling. `Decimal` does have the snapshots disabled in the test stories though.

Reply:
you're right - I checked Playwright tests and couldn't find one for Optional, but it should be covered by Chromatic as the [relevant story](https://github.com/Sage/carbon/blob/master/src/components/decimal/decimal.stories.tsx#L174) doesn't seem to have it disabled. I'll therefore simply remove this as suggested 👍